### PR TITLE
chore: refactor useFetchAssets to make it more readable

### PR DESCRIPTION
**Pull request by JackShort**
* note didn't refactor the old fetch code since it will be removed

### CODE REVIEW

File: src/nft/utils/bag.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    return { nextBagStatus: BagStatus.CONFIRM_REVIEW, lockBag: false }
+  }
+
+  if (shouldRefetchCalldata) {
+    return { nextBagStatus: BagStatus.CONFIRM_QUOTE, lockBag: false }
+  }
+
+  return { nextBagStatus: BagStatus.CONFIRMING_IN_WALLET, lockBag: true }
```
Comment (cbachmeier) [inline src/nft/utils/bag.ts:53]: Are there cases where the BagStatuses and lockBag state differ from what's here? You could create an enum/map mapping status to lockBag state so you don't have to track them separately.

File: src/nft/utils/bag.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  const { hasPriceAdjustment, updatedAssets } = compareAssetsWithTransactionRoute(wishAssetsToBuy, route)
+  const shouldRefetchCalldata = hasPriceAdjustment && purchasingWithErc20
+
+  const { unchanged, priceChanged, unavailable } = filterUpdatedAssetsByState(updatedAssets)
+
+  const hasReviewedAssets = unchanged.length > 0
+  const hasAssetsInReview = priceChanged.length > 0
+  const hasUnavailableAssets = unavailable.length > 0
+
+  const hasAssets = hasReviewedAssets || hasAssetsInReview || hasUnavailableAssets
```
Comment (cbachmeier) [inline src/nft/utils/bag.ts:70]: very nit: could also just be !!updatedAssets.length

**Review by cbachmeier — APPROVED** [review]
A lot cleaner and works well.

File: src/nft/utils/bag.ts (hunk 2 above)
Comment (JackShort) [reply src/nft/utils/bag.ts:70]: its crazy how true that is

File: src/nft/utils/bag.ts (hunk 1 above)
Comment (JackShort) [reply src/nft/utils/bag.ts:53]: crazier how none of this is actually necessary when one state only returns false lmao

**Pull request merged**